### PR TITLE
New format of source links

### PR DESCRIPTION
**Pull request by romanowski**
Now we use liquid templates to generate link to source file.
We use our custom-build support for handling this instead of one from dokka.

This commit also adds predefined templates for github and gitlab as well ass 'revision setting' that allows users to specify revision of documented code (and source link)

### CODE REVIEW

**Comment by romanowski** [comment]
cc @BarkingBad @pikinier20 @Kordyjan

**Review by BarkingBad — APPROVED** [review]
Looks ok

**Review by abgruszecki — COMMENTED** [review]
LGTM.

File: scala3doc/src/dotty/dokka/SourceLinks.scala [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  )
+
+  def githubTemplate(organization: String, repo: String)(revision: String) =
+    s"""https://github.com/$organization/$repo/{{ operation | replace: "view", "blob" }}/$revision/{{ path }}#L{{ line }}""".stripMargin
+
+  def gitlabTemplate(organization: String, repo: String)(revision: String) =
+    s"""https://gitlab.com/$organization/$repo/-/{{ operation | replace: "view", "blob" }}/$revision/{{ path }}#L{{ line }}"""
+
+
+  private def parseLinkDefinition(s: String): Option[SourceLink] = ???
```
Comment (MarcinAman) [inline scala3doc/src/dotty/dokka/SourceLinks.scala:27]: I dont think it is used :D

**Pull request merged**